tests/test_TFNetworkRecLayer.py

```diff
@@ -5294,6 +5294,79 @@ def test_trafo_search_lm():
       assert out_seq_lens[i] == input_seq_lens[i] * 3 // 2  # we constructed the 'end' layer that way
       assert all(out_seqs[i, :input_seq_lens[i]] == input_seqs[i, :input_seq_lens[i]])
 
+def test_cumulated_attention_weights_search():
+  rnd = numpy.random.RandomState(42)
+  beam_size = 5
+  dim = 7
+
+  # Config works during training, but building graph raises exception during search:
+  # Trying to reshape input tensor with n values into tensor with n * beam_size values
+  net_dict = {
+    'source_embed': { 'class': 'linear', 'activation': None, 'n_out': dim},
+    'output': {   'class': 'rec',
+                  'from': [],
+                  'max_seq_len': "max_len_from('base:source_embed') * 3",
+                  'target': 'classes',
+                  'unit': {
+                      'target_embed': { 'class': 'linear', 'activation': None, 'from': ['prev:output'], 'n_out': dim},
+                      'att_energy': { 'class': 'dot', 'from': ['base:source_embed', 'target_embed'], 'red1': -1, 'red2': -1, 'var1': 'T', 'var2': 'T?', 'add_var2_if_empty': False},
+                      'cum_att_energy': {'class': 'combine', 'kind': 'add', 'from': ['prev:att_energy', 'att_energy'], 'initial_output': 'att_energy'},
```

Comment:
Does `'initial_output': 'att_energy'` make sense? I think not. I wonder that this works (but maybe actually that's the reason why it does not). `initial_output` specifies the initial output. I.e. when you access `prev: cum_att_energy` in the first frame, it specifies the output. But as you never do that anyway, it's probably ignored anyway. But then better remove this.